api/queue/work/model_service_deployment_test.go

```diff
@@ -128,6 +234,7 @@ func TestExecuteDeployment(t *testing.T) {
 			},
 			deploymentStorage: func() *mocks.DeploymentStorage {
 				mockStorage := &mocks.DeploymentStorage{}
+				mockStorage.On("GetLatestDeployment", mock.Anything, mock.Anything).Return(nil, gorm.ErrRecordNotFound)
```

Comment:
instead of adding this to all, perhaps create a function that returns a default behaviour used by most and construct from there and override if needed? i.e

```
func createMockDeploymentStorage() *mocks.VersionEndpointStorage {
	mockStorage := &mocks.VersionEndpointStorage{}
	mockStorage.On("GetLatestDeployment", mock.Anything, mock.Anything).Return(nil, gorm.ErrRecordNotFound)
	mockStorage.On("Save", mock.Anything).Return(nil)
	mockStorage.On("OnDeploymentSuccess", mock.Anything).Return(nil)
	return mockStorage
}

```

Reply:
Hmm that's true, that'll make things cleaner. Let me try refactoring this.